buffer/buffer.go

```diff
@@ -164,6 +164,19 @@ func (b *Buffer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	outReq := b.copyRequest(req, body, totalSize)
 
+	defer func() {
+		if err := recover(); err != nil {
+			if err == http.ErrAbortHandler {
+				b.log.Error("vulcand/oxy/buffer: failed to read response, err: %v", err)
+
+				b.errHandler.ServeHTTP(w, req, err.(error))
+				return
+			}
+
+			panic(err)
```

Comment:
Not sure about that.
Why do you not write the error inside the response?

Reply:
The fact that the `Write` method of `bufferWriter` doesn't return an error prevents `ErrAbortHandler` from being raised.
The idea is to fix the `Write` method and specifically handles `ErrAbortHandler` by returning a 500 status code to avoid breaking changes.
For other errors, a panic is raised to ensure that existing panic behaviors remain unchanged.

After giving it some thought, I wonder if there might be impacts I didn't anticipate. Perhaps an approach like this (https://github.com/vulcand/oxy/pull/247) might be safer.